test/controllers/VerifiedEmailControllerSpec.scala

```diff
@@ -229,14 +229,124 @@ class VerifiedEmailControllerSpec extends SpecBase with MockAuthConnector {
+        status(result) mustBe OK
+        contentAsJson(result) mustBe Json.obj("address" -> testAddress, "timestamp" -> s"${testTime.toString}Z")
+        verifyNoInteractions(mockSubscriptionInfoService)
+      }
+    }
+
+    "return the email and call SUB09 if the data is not stored in the cache and " +
+      "also store the response into the cache" in new Setup {
+        when(mockEmailRepository.get(any()))
+          .thenReturn(Future.successful(None))
+          .thenReturn(Future.successful(Some(NotificationEmail(testAddress, testTime, None))))
+
+        when(mockSubscriptionInfoService.getSubscriberInformation(any())).thenReturn(
+          Future.successful(
+            Some(NotificationEmail(testAddress, testTime, None))
+          )
+        )
+
+        when(mockEmailRepository.set(any(), any())).thenReturn(Future.successful(SuccessfulEmail))
+
+        val request: FakeRequest[AnyContentAsJson] =
+          FakeRequest(GET, retrieveVerifiedEmailThirdPartyRoute).withJsonBody(
+            Json.obj("eori" -> TEST_EORI_VALUE)
+          )
+
+        running(app) {
+          val result = route(app, request).value
+          status(result) mustBe OK
+          contentAsJson(result) mustBe Json.obj("address" -> testAddress, "timestamp" -> s"${testTime.toString}Z")
+        }
+      }
+
+    "return InternalServerError if the write did not succeed when retrieving email from SUB09" in new Setup {
+      when(mockEmailRepository.get(any()))
+        .thenReturn(Future.successful(None))
+        .thenReturn(Future.successful(Some(NotificationEmail(testAddress, testTime, None))))
+
+      when(mockSubscriptionInfoService.getSubscriberInformation(any())).thenReturn(
+        Future.successful(
+          Some(NotificationEmail(testAddress, testTime, None))
+        )
+      )
+
+      when(mockEmailRepository.set(any(), any())).thenReturn(Future.successful(FailedToRetrieveEmail))
+
+      val request: FakeRequest[AnyContentAsJson] = FakeRequest(GET, retrieveVerifiedEmailThirdPartyRoute).withJsonBody(
+        Json.obj("eori" -> TEST_EORI_VALUE)
+      )
+
+      running(app) {
+        val result = route(app, request).value
+        status(result) mustBe INTERNAL_SERVER_ERROR
+      }
+    }
+
+    "return 400 with malformed request" in new Setup {
+      when(mockEmailRepository.get(any()))
+        .thenReturn(Future.successful(None))
+        .thenReturn(Future.successful(Some(NotificationEmail(testAddress, testTime, None))))
+
+      when(mockSubscriptionInfoService.getSubscriberInformation(any())).thenReturn(
+        Future.successful(
+          Some(NotificationEmail(testAddress, testTime, None))
+        )
+      )
+
+      when(mockEmailRepository.set(any(), any())).thenReturn(Future.successful(SuccessfulEmail))
+
+      val request: FakeRequest[AnyContentAsJson] = FakeRequest(GET, retrieveVerifiedEmailThirdPartyRoute).withJsonBody(
+        Json.obj("invalidkey" -> TEST_EORI_VALUE)
+      )
```

Comment:
There is normally a new line after the requests ) to space it out from the running, like on line 309